test/unit/puppetlabs/services/master/master_service_test.clj

```diff
@@ -70,3 +69,56 @@
                   (test-path! "cert_inventory" "target/master-service-test/inventory.txt"))))))
         (finally
           (fs/delete-dir test-dir))))))
+
+; This ensures that a test failure will be logged for the version check
+; test if the master for some reason stops calling into the version check
+; library
+(def version-check-success
+  (atom false))
+
+(defn version-check-test-fn
+  [request-values update-server-url]
+  (is (= {:group-id "puppets" :artifact-id "yoda"} (:product-name request-values)))
+  (is (= "http://notarealurl/" update-server-url))
+  (swap! version-check-success not))
+
+; This exists to prevent the master from creating SSL files, and thus make the test run
+; faster
+(defn dummy-retrieve-cert-function
+  [_]
+  (println "Do nothing"))
+
+(deftest version-check-test
+  (testing "master calls into the dujour version check library using the correct values"
+    (with-redefs
+
+      [version-check/check-for-updates! version-check-test-fn
+       ; Redefine the CA functions to ensure that the master does not create
```

Comment:
I'm good with mocking these here since this functionality isn't related to the test.